libr/cons/dietline.c

```diff
@@ -68,11 +71,27 @@ static void kill_word() {
 	if (I.buffer.index >= I.buffer.length) {
 		I.buffer.length = I.buffer.index;
 	}
-	memmove (I.buffer.data + I.buffer.index, I.buffer.data + i,
-			I.buffer.length - i + 1);
+	len = i - I.buffer.index + 1;
+	free (I.clipboard);
+	I.clipboard = r_str_ndup(I.buffer.data + I.buffer.index, len);
+	memmove (I.buffer.data + I.buffer.index, I.buffer.data + i, len);
 	I.buffer.length = strlen (I.buffer.data);
 }
 
+static void yank() {
+	int len, dist;
+	char *cursor;
+	if (I.clipboard) {
```

Comment:
you can return earlier and avoid some nesting in here

Reply:
Do you mean checking for `I.clipboard` in the `case 25` statement rather than in the function?